marconi/test/Helpers.hs

```diff
@@ -0,0 +1,243 @@
+  -- era.
+  socketPathAbs <- getSocketPathAbs conf tn
+  let epochSlots = C.EpochSlots 21600
+      localNodeConnectInfo =
+        C.LocalNodeConnectInfo
+          { C.localConsensusModeParams = C.CardanoModeParams epochSlots
+          , C.localNodeNetworkId = getNetworkId tn
+          , C.localNodeSocketPath = socketPathAbs
+          }
+
+  pure (localNodeConnectInfo, conf, tn)
+
+getNetworkId :: TN.TestnetRuntime -> C.NetworkId
+getNetworkId tn = C.Testnet $ C.NetworkMagic $ fromIntegral (TN.testnetMagic tn)
+
+getSocketPathAbs :: (MonadTest m, MonadIO m) => TC.Conf -> TN.TestnetRuntime -> m FilePath
+getSocketPathAbs conf tn = do
+  let tempAbsPath = TC.tempAbsPath conf
+  socketPath <- IO.sprocketArgumentName <$> H.headM (TN.nodeSprocket <$> TN.bftNodes tn)
+  socketPathAbs <- H.note =<< (liftIO $ IO.canonicalizePath $ tempAbsPath </> socketPath)
+  pure socketPathAbs
+
+getPoolSocketPathAbs :: (MonadTest m, MonadIO m) => TC.Conf -> TN.TestnetRuntime -> m FilePath
+getPoolSocketPathAbs conf tn = do
+  let tempAbsPath = TC.tempAbsPath conf
+  socketPath <- IO.sprocketArgumentName <$> H.headM (TN.poolNodeSprocket <$> TN.poolNodes tn)
+  socketPathAbs <- H.note =<< (liftIO $ IO.canonicalizePath $ tempAbsPath </> socketPath)
+  pure socketPathAbs
+
+--
+
+readAs :: (C.HasTextEnvelope a, MonadIO m, MonadTest m) => C.AsType a -> FilePath -> m a
+readAs as path = do
+  path' <- H.note path
+  H.leftFailM . liftIO $ C.readFileTextEnvelope as path'
+
+-- * Talk to node
+
+emptyTxBodyContent :: C.Lovelace -> C.ProtocolParameters -> C.TxBodyContent C.BuildTx C.AlonzoEra
+emptyTxBodyContent fee pparams = C.TxBodyContent
+  { C.txIns              = []
+  , C.txInsCollateral    = C.TxInsCollateralNone
+  , C.txInsReference     = C.TxInsReferenceNone
+  , C.txOuts             = []
+  , C.txTotalCollateral  = C.TxTotalCollateralNone
+  , C.txReturnCollateral = C.TxReturnCollateralNone
+  , C.txFee              = C.TxFeeExplicit C.TxFeesExplicitInAlonzoEra fee
+  , C.txValidityRange    = (C.TxValidityNoLowerBound, C.TxValidityNoUpperBound C.ValidityNoUpperBoundInAlonzoEra)
+  , C.txMetadata         = C.TxMetadataNone
+  , C.txAuxScripts       = C.TxAuxScriptsNone
+  , C.txExtraKeyWits     = C.TxExtraKeyWitnessesNone
+  , C.txProtocolParams   = C.BuildTxWith $ Just pparams
+  , C.txWithdrawals      = C.TxWithdrawalsNone
+  , C.txCertificates     = C.TxCertificatesNone
+  , C.txUpdateProposal   = C.TxUpdateProposalNone
+  , C.txMintValue        = C.TxMintNone
+  , C.txScriptValidity   = C.TxScriptValidityNone
+  }
+
+getAlonzoProtocolParams :: (MonadIO m, MonadTest m) => C.LocalNodeConnectInfo C.CardanoMode -> m C.ProtocolParameters
+getAlonzoProtocolParams localNodeConnectInfo = H.leftFailM . H.leftFailM . liftIO
+  $ C.queryNodeLocalState localNodeConnectInfo Nothing
+  $ C.QueryInEra C.AlonzoEraInCardanoMode
+  $ C.QueryInShelleyBasedEra C.ShelleyBasedEraAlonzo C.QueryProtocolParameters
```

Comment:
Out of interest, why are we using `ShelleyBasedEraAlonzo` instead of `Babbage`?

Reply:
So this had no other reason than that Alonzo was what I started out with :). But now that I tried to switch to Babbage, then it appears that the cardano-testnet:Testnet.Cardano module doesn't support it currently. There is also the Testnet.Babbage module beside it, but that would need the [foldBlocks fix](https://github.com/input-output-hk/cardano-node/pull/4680) to make it work.